polling/nf_configuration_test.go

```diff
@@ -0,0 +1,290 @@
+// SPDX-FileCopyrightText: 2025 Canonical Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+/*
+ * NF Polling Unit Tests
+ *
+ */
+
+package polling
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/omec-project/amf/factory"
+	"github.com/omec-project/openapi/models"
+	"github.com/omec-project/openapi/nfConfigApi"
+)
+
+func makeAccessMobilityConfig(mcc, mnc, sst string, sd string, tacs []string) nfConfigApi.AccessAndMobility {
+	sstUint64, err := strconv.ParseUint(sst, 10, 8)
+	if err != nil {
+		panic("invalid SST value: " + sst)
+	}
+	sstint := int32(sstUint64)
+	return nfConfigApi.AccessAndMobility{
+		PlmnId: nfConfigApi.PlmnId{
+			Mcc: mcc,
+			Mnc: mnc,
+		},
+		Snssai: nfConfigApi.Snssai{
+			Sst: sstint,
+			Sd:  &sd,
+		},
+		Tacs: tacs,
+	}
+}
+
+func TestStartPollingService_Success(t *testing.T) {
+	ctx := t.Context()
+	originalFetchAccessAndMobilityConfig := fetchAccessAndMobilityConfig
+	originalFactoryConfig := factory.AmfConfig
+	defer func() {
+		fetchAccessAndMobilityConfig = originalFetchAccessAndMobilityConfig
+		factory.AmfConfig = originalFactoryConfig
+	}()
+	factory.AmfConfig = factory.Config{
+		Configuration: &factory.Configuration{
+			SupportTAIList:  []models.Tai{},
+			PlmnSupportList: []factory.PlmnSupportItem{},
+		},
+	}
+
+	expectedConfig := []nfConfigApi.AccessAndMobility{
+		{
+			PlmnId: nfConfigApi.PlmnId{Mcc: "001", Mnc: "01"},
+			Snssai: nfConfigApi.Snssai{Sst: 1},
+			Tacs:   []string{"1"},
+		},
+	}
+
+	fetchAccessAndMobilityConfig = func(poller *nfConfigPoller, pollingEndpoint string) ([]nfConfigApi.AccessAndMobility, error) {
+		return expectedConfig, nil
+	}
+	regChan := make(chan []nfConfigApi.AccessAndMobility, 1)
+	updateCtxChan := make(chan []nfConfigApi.AccessAndMobility, 1)
+	go StartPollingService(ctx, "http://dummy", regChan, updateCtxChan)
+	time.Sleep(initialPollingInterval)
+
+	select {
+	case result := <-updateCtxChan:
+		if !reflect.DeepEqual(result, expectedConfig) {
+			t.Errorf("Expected %+v, got %+v", expectedConfig, result)
```

Comment:
Make this consistent, in some cases it is uppercased and in other it is not across this file
For example:

```suggestion
			t.Errorf("expected %+v, got %+v", expectedConfig, result)
```